src/elements/toniq-slider/toniq-slider.element.ts

```diff
@@ -0,0 +1,407 @@
+            lowerLabel.style.left = `${
+                lowerLabelOffset < lowerSlider.getBoundingClientRect().left
+                    ? lowerSlider.getBoundingClientRect().left
+                    : lowerLabelOffset
+            }px`;
+
+            const upperLabelOffset =
+                progress.getBoundingClientRect().right - upperLabel.offsetWidth / 2;
+            upperLabel.style.left = `${
+                upperLabelOffset + upperLabel.clientWidth >
+                upperSlider.getBoundingClientRect().right
+                    ? upperSlider.getBoundingClientRect().right - upperLabel.clientWidth
+                    : upperLabelOffset
+            }px`;
+
+            // Plus 16px (8px half of font size + 8px spacing from thumb to label based on design)
+            lowerLabel.style.top = `${progress.getBoundingClientRect().top + 16}px`;
+            upperLabel.style.top = `${progress.getBoundingClientRect().top + 16}px`;
+
+            const valueOffset = 10;
+            if (
+                lowerLabel.getBoundingClientRect().right + valueOffset >=
+                    upperLabel.getBoundingClientRect().left ||
+                upperLabel.getBoundingClientRect().left <= lowerLabel.getBoundingClientRect().right
+            ) {
+                const lowerLabelOffset =
+                    progress.getBoundingClientRect().left -
+                    lowerLabel.getBoundingClientRect().width;
+                const upperLabelOffset = progress.getBoundingClientRect().right;
+                const lowerLabelMin = lowerSlider.getBoundingClientRect().left;
+
+                const lowerLabelMax =
+                    lowerSlider.getBoundingClientRect().right -
+                    upperLabel.getBoundingClientRect().width -
+                    lowerLabel.getBoundingClientRect().width -
+                    valueOffset;
+
+                const upperLabelMin =
+                    upperSlider.getBoundingClientRect().left +
+                    lowerLabel.getBoundingClientRect().width +
+                    valueOffset;
+                const upperLabelMax =
+                    upperSlider.getBoundingClientRect().right -
+                    upperLabel.getBoundingClientRect().width;
+
+                lowerLabel.style.left = `${
+                    lowerLabelOffset < lowerLabelMin
+                        ? lowerLabelMin
+                        : lowerLabelOffset > lowerLabelMax
+                        ? lowerLabelMax
+                        : lowerLabelOffset
+                }px`;
+                upperLabel.style.left = `${
+                    upperLabelOffset < upperLabelMin
+                        ? upperLabelMin
+                        : upperLabelOffset > upperLabelMax
+                        ? upperLabelMax
+                        : upperLabelOffset
+                }px`;
+            }
+        }
+
+        function onSliderCreated(slider: Element) {
```

Comment:
This function should be deleted; everything done in here should be assigned in the html as properties and listeners.